ALVRClient/GlobalSettings.swift

```diff
@@ -3,11 +3,54 @@
 //
 
 import Foundation
+import SwiftUI
 
-class GlobalSettings: ObservableObject {
-    static let shared = GlobalSettings()
+struct GlobalSettings: Codable {
+    var keepSteamVRCenter: Bool
+    var showHandsOverlaid: Bool
     
-    // TODO: configuration persistence
-    var keepSteamVRCenter = true
-    var showHandsOverlaid = false
+    init(keepSteamVRCenter: Bool, showHandsOverlaid: Bool) {
+        self.keepSteamVRCenter = keepSteamVRCenter
+        self.showHandsOverlaid = showHandsOverlaid
+    }
+}
+
+extension GlobalSettings {
+    static let sampleData: GlobalSettings =
+    GlobalSettings(keepSteamVRCenter: true, showHandsOverlaid: true)
+}
+
+@MainActor
+class GlobalSettingsStore: ObservableObject {
+    @Published var settings: GlobalSettings = GlobalSettings(keepSteamVRCenter: false, showHandsOverlaid: false)
+    
+    private static func fileURL() throws -> URL {
+        try FileManager.default.url(for: .documentDirectory,
+                                    in: .userDomainMask,
+                                    appropriateFor: nil,
+                                    create: true)
+        .appendingPathComponent("globalsettings.data")
+    }
+    
+    func load() async throws {
+        let task = Task<GlobalSettings, Error> {
+            let fileURL = try Self.fileURL()
+            guard let data = try? Data(contentsOf: fileURL) else {
+                return GlobalSettings(keepSteamVRCenter: false, showHandsOverlaid: false)
+            }
+            let globalSettings = try JSONDecoder().decode(GlobalSettings.self, from: data)
+            return globalSettings
+        }
+        let settings = try await task.value
+        self.settings = settings
```

Comment:
The Task can be removed, since the `load` method is async. Moreover, the load method doesn't need to be async since everything in the task is synchronous.

Reply:
Updated load function

Reply:
The same applies to `save` as well. Sorry, I should have been clearer

Reply:
Updated save function too